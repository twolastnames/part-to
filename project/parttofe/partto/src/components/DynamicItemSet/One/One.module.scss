.one {
  display: flex;
  width: 100%;
}

.content {
  padding: 6px;
  width: 100%;
  justify-self: center;
}
