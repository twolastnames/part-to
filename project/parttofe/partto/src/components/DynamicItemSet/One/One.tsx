import React from "react";

import classes from "./One.module.scss";
import { Item } from "../DynamicItemSet";
import { ButtonSet } from "../ButtonSet/ButtonSet";

export interface OneProps {
  item: Item;
}

export function One({ item }: OneProps) {
  return (
    <div className={classes.one} data-testid="One">
      <div className={classes.content}>{item.detailView}</div>
      <ButtonSet operations={item.itemOperations} />
    </div>
  );
}
