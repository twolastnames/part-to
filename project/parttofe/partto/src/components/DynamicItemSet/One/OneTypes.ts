import { Item } from "../DynamicItemSetTypes";

export interface OneProps {
  item: Item;
}
