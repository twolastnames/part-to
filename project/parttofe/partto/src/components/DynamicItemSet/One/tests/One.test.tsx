import React from "react";
import { expect, test } from "@jest/globals";
import { render, screen } from "@testing-library/react";
import { One } from "../One";

test("snapshot", () => {
  render(
    <One
      item={{
        listView: <div>invisible</div>,
        detailView: <div>shown</div>,
        itemOperations: [],
      }}
    />,
  );

  const component = screen.getByTestId("One");
  expect(component).toMatchSnapshot();
});
