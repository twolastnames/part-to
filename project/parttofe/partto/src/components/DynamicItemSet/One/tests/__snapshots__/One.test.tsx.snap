// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  class="one"
  data-testid="One"
>
  <div
    class="content"
  >
    <div>
      shown
    </div>
  </div>
  <div
    class="buttonSet"
    data-testid="ButtonSet"
  />
</div>
`;
